src/main/java/com/vfasad/controller/reports/CommonReportController.java

```diff
@@ -112,8 +125,8 @@ public ModelAndView monthReport(@PathVariable int year,
         modelAndView.addObject("areaByTeamId", areaByTeamId);
 
         modelAndView.addObject("usersByTeamId", userService.getUsersByTeamId());
-
-        modelAndView.addObject("options", optionService.getOptionsMap());
+        Map<String, String> options = reportService.findByYearAndMonthReportOptions(year, month);
+        modelAndView.addObject("options",  options.size() ==0 ? optionService.getOptionsMap() : options);
```

Comment:
Formatting, spaces.

Reply:
The real condition to use generic options is report state, right? Maybe it would be more obvious to use it to decide?